ros/angel_system_nodes/angel_system_nodes/audio_player.py

```diff
@@ -0,0 +1,85 @@
+import copy
+import threading
+import time
+
+import simpleaudio as sa
+
+import rclpy
+from rclpy.node import Node
+from angel_msgs.msg import HeadsetAudioData
+
+
+class AudioPlayer(Node):
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+
+        self.declare_parameter("audio_topic", "HeadsetAudioData")
+
+        self._audio_topic = self.get_parameter("audio_topic").get_parameter_value().string_value
+
+        self.subscription = self.create_subscription(
+            HeadsetAudioData,
+            self._audio_topic,
+            self.listener_callback,
+            1)
+
+        self.audio_stream = bytearray()
+        self.audio_player_lock = threading.RLock()
+        self.t = threading.Thread()
+
+        self.audio_stream_duration = 0.0
+        self.playback_duration = 1.0
+
+    def listener_callback(self, msg):
+        """
+        Add the audio to the cumulative audio stream and spawns a new playback
+        thread if the cumulative recording has reached more than a second and
+        there is no ongoing playback.
+        """
+        self.audio_stream.extend(msg.data)
+        self.audio_stream_duration += msg.sample_duration
+
+        if self.audio_stream_duration >= self.playback_duration and not(self.t.is_alive()):
+            # Make a copy of the current data so we can play it back
+            # while more data is accumulated
+            audio_data = copy.deepcopy(self.audio_stream)
```

Comment:
Baring a complication I don't realize, I don't think you need to deep-copy here (deep copies are expensive). You are throwing away the instance we just deep copied right after this by assigning the variable a new `bytearray` instance.